src/test/scala/nl.knaw.dans.easy.multideposit/BlackBoxSpec.scala

```diff
@@ -19,37 +19,63 @@ import java.nio.file.{ Files, Path, Paths }
 import javax.naming.directory.{ Attributes, BasicAttribute, BasicAttributes }
 
 import org.scalamock.scalatest.MockFactory
-import org.scalatest.BeforeAndAfter
-import resource._
+import resource.managed
 
 import scala.collection.JavaConverters._
-import scala.util.{ Failure, Success }
-import scala.xml.transform.{ RewriteRule, RuleTransformer }
+import scala.util.{ Failure, Properties, Success }
 import scala.xml.{ Elem, Node, NodeSeq, XML }
+import scala.xml.transform.{ RewriteRule, RuleTransformer }
 
-class BlackBoxSpec extends UnitSpec with BeforeAndAfter with MockFactory with CustomMatchers {
+// Note to developers: this classes uses shared tests as described in
+// http://www.scalatest.org/user_guide/sharing_tests
+class BlackBoxSpec extends UnitSpec with MockFactory with CustomMatchers {
 
   private val formatsFile: Path = testDir.resolve("formats.txt")
   private val allfields = testDir.resolve("md/allfields").toAbsolutePath
   private val invalidCSV = testDir.resolve("md/invalidCSV").toAbsolutePath
 
-  before {
+  /*
+    Note to future developers:
+    We're sharing tests in this BlackBoxSpec to prevent as much test duplication as possible.
+    However, a test like `"allfields" should behave like allfieldsSpec()`, combined with a
+    `BeforeAndAfterAll` will always first execute the setup stuff from `allfieldsSpec()` and only
+    then run the `beforeAll` and `afterAll` functions. Other traits like `BeforeAndAfter` and
+    `BeforeAndAfterEach` are isomorphic.
+    By not using these traits but manually defining and calling `beforeAll` we make sure this
+    function is called before the shared tests are set up or ran.
+   */
+  def beforeAll(): Unit = {
```

Comment:
Might be. The issue is that ScalaTest is doing discovery on the tests before they are actually executed. This means that some setup stuff is executed twice: once in the discovery phase, once during the actual test itself. Now the issue I ran into yesterday with this has to do with the part that is executed during discovery. Apparently that includes stuff that is defined within the functions below, but outside the tests that are in the functions (part of the setup). This part is ran before the functions `beforeAll`, etc. That's why we need the manual method call here. Note that we leverage the fact that tests are ran in sequence in ScalaTest, rather than jUnit, which doesn't have a particular order described to its tests.